Code/GraphMol/Chirality.cpp

```diff
@@ -2659,6 +2660,70 @@ void removeNonExplicit3DChirality(ROMol &mol) {
   }
 }
 
+void addStereoAnnotations(ROMol &mol) {
+  auto sgs = mol.getStereoGroups();
+  assignStereoGroupIds(sgs);
+  std::vector<unsigned int> doneAts(mol.getNumAtoms(), 0);
+  for (const auto &sg : sgs) {
+    for (const auto atom : sg.getAtoms()) {
+      if (doneAts[atom->getIdx()]) {
+        BOOST_LOG(rdWarningLog) << "Warning: atom " << atom->getIdx()
+                                << " is in more than one stereogroup. Only the "
+                                   "label from the first group will be used."
+                                << std::endl;
+        continue;
+      }
+      std::string lab;
+      std::string cip;
+      if (sg.getGroupType() == StereoGroupType::STEREO_ABSOLUTE) {
+        atom->getPropIfPresent(common_properties::_CIPCode, cip);
+      }
+      switch (sg.getGroupType()) {
+        case StereoGroupType::STEREO_ABSOLUTE:
+          lab = "abs";
+          break;
+        case StereoGroupType::STEREO_OR:
+          lab = (boost::format("or%d") % sg.getWriteId()).str();
```

Comment:
Would it be useful to allow to customize these, e.g.:

```
/// Formatting to use for Stereo labels, as printf format strings
struct StereoLabelFormats {
    // %s=chirality (R,S,etc)
    std::string absLabel = "abs (%s)"; 
    // %d=stereo group id; %s=chirality (R,S,etc)
    std::string andLabel = "and%d (%s)";
    std::string orLabel = "or%d (%s)";
    // %s=E,Z,etc
    std::string dblBondLabel = "%s"
};

addStereoAnnotations(ROMol &mol, StereoLabelFormats labels=StereoLabelFormats())
```

Adding the CIP label to AND & OR groups is intentional: I know we don't use it, but maybe someone wants it?

Also, maybe instead of `%s` or `%d` we could/should use custom formatters like `%chirality` and `%groupid` ?

Reply:
`Adding the CIP label to AND & OR groups is intentional: I know we don't use it, but maybe someone wants it?`

No one has the ability to access this before, so afaik no one has used it. Greg would know if someone has requested it, but my guess is no.

Reply:
No one has requested it yet, but It seems like it can't hurt to add the flexibility.
The PR has been revised to allow the format of the labels to be specified using some very simple templating